### PR TITLE
CI/CD 세팅

### PR DESCRIPTION
**Pull request by mingmong0018**
# 변경 사항
- 테스트 커버리지 리포트를 받기 위해 jacoco 세팅
- application.yml에 중요정보 시스템환경변수 사용
- 애플리케이션 이미지 빌드를 위해 Dockerfile 생성
- init.sql을 이용하여 초기 데이터 세팅을 적용한 mysql 이미지 빌드를 위해 Dockerfile 생성
- redis, mysql, app 컨테이너를 관리하기 위한 docker-compose.yml 생성

### CODE REVIEW

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/479671fdf0591e03e1558d6cf7662c154f0a7835/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/479671fdf0591e03e1558d6cf7662c154f0a7835/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/5c0962723bd95d1a935cddd6d0fb766420cf30a2/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/5c0962723bd95d1a935cddd6d0fb766420cf30a2/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/5c0962723bd95d1a935cddd6d0fb766420cf30a2/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/5c0962723bd95d1a935cddd6d0fb766420cf30a2/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/abf5760ef69679b4dd1caa3dede7fd01c4c6ca11/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/abf5760ef69679b4dd1caa3dede7fd01c4c6ca11/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/0bb9f4d52f2a0728a0dcf8647621540c40c85410/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/0bb9f4d52f2a0728a0dcf8647621540c40c85410/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/1ca4f06151c81b5a36d9f0c9eb7a847e9e0da44f/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/1ca4f06151c81b5a36d9f0c9eb7a847e9e0da44f/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/55979c034474a5f2e571f6b2dfe98488f7020655/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/55979c034474a5f2e571f6b2dfe98488f7020655/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/78728fe775318e89d8f7e0deb248327f177a1fc6/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/78728fe775318e89d8f7e0deb248327f177a1fc6/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/8c062a0ab469964f9c3554ef7119e11c290cf430/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/8c062a0ab469964f9c3554ef7119e11c290cf430/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/2cced127f9d6c19087a567a4b0c4888368a49db8/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/2cced127f9d6c19087a567a4b0c4888368a49db8/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/67e4aff5ce885d00b178318ea1cdafce4fcadcb7/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/67e4aff5ce885d00b178318ea1cdafce4fcadcb7/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/96fa4cf75269b5005ad81683c2df2cd776758bd6/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/96fa4cf75269b5005ad81683c2df2cd776758bd6/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/79ea35ed3677e34fa3e1c0b119748847759b59ba/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/79ea35ed3677e34fa3e1c0b119748847759b59ba/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/7b1dba08305074d53115d1d999ea23939b231960/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/7b1dba08305074d53115d1d999ea23939b231960/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

**Comment by mingmong0018** [comment]
### 테스트 커버리지 리포트
|File|Coverage [0%]|:x:|
|:-|:-:|:-:|
|[RedisTokenConfig.java](https://github.com/f-lab-edu/OnlineMarket/blob/717aeca53798776a8de51cbb67a2f81655ea1875/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fconfig%2FRedisTokenConfig.java)|0%|:x:|
|[RedisProperties.java](https://github.com/f-lab-edu/OnlineMarket/blob/717aeca53798776a8de51cbb67a2f81655ea1875/src%2Fmain%2Fjava%2Fcom%2Fmarket%2Fglobal%2Fproperties%2FRedisProperties.java)|0%|:x:|

|Total Project Coverage|55.28%|:x:|
|:-|:-:|:-:|

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ server:
...
     datasource:
         driver-class-name: com.mysql.cj.jdbc.Driver
-        url: jdbc:mysql://localhost:3306/Market?serverTimezone=UTC&useLegacyDatetimeCode=false
+        url: jdbc:mysql://223.130.142.30:3306/Market?serverTimezone=UTC&useLegacyDatetimeCode=false
         username: ${SPRING_DATASOURCE_USERNAME}
         password: ${SPRING_DATASOURCE_PASSWORD}
     data:
         redis:
-            host: localhost
+            host: 223.130.142
```
Comment (f-lab-bradley) [inline src/main/resources/application.yml]: redis 서버는 따로 만드신거에요???

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -4,14 +4,14 @@ server:
 spring:
     datasource:
         driver-class-name: com.mysql.cj.jdbc.Driver
-        url: jdbc:mysql://223.130.142.30:3306/Market?serverTimezone=UTC&useLegacyDatetimeCode=false
-        username: ${SPRING_DATASOURCE_USERNAME}
-        password: ${SPRING_DATASOURCE_PASSWORD}
+        url: jdbc:mysql://${DATABASE_HOST}:${DATABASE_PORT}/Market?serverTimezone=UTC&useLegacyDatetimeCode=false
```
Comment (f-lab-bradley) [inline src/main/resources/application.yml:4]: 환경변수 방식으로 수정한거 굉장히 잘 하셨어요 ㅎㅎ

File: src/main/java/com/market/global/config/RedisTokenConfig.java [hunk 3]
```diff
@@ -19,11 +18,7 @@ public class RedisTokenConfig {
 
 	@Bean
 	public RedisConnectionFactory redisConnectionFactory() {
-		RedisStandaloneConfiguration config = new RedisStandaloneConfiguration();
-		config.setHostName(redisProperties.getHost());
-		config.setPort(redisProperties.getPort());
-		config.setPassword(redisProperties.getPassword());
-		return new LettuceConnectionFactory(config);
+		return new LettuceConnectionFactory(redisProperties.getHost(), redisProperties.getPort());
```
Comment (f-lab-bradley) [inline src/main/java/com/market/global/config/RedisTokenConfig.java:21]: config 방식으로 해도 좋을 것 같은데 왜 굳이 두 개만 설정 하도록 했어요????
redis 는 connection pool 이 없나요??

File: src/main/java/com/market/global/properties/RedisProperties.java [hunk 4]
```diff
@@ -13,5 +13,4 @@
 public class RedisProperties {
 	private String host;
 	private int port;
-	private String password;
```
Comment (f-lab-bradley) [inline src/main/java/com/market/global/properties/RedisProperties.java:16]: password 는 왜 뺐을까요??

**Review by f-lab-bradley — COMMENTED** [review]
서버 띄우는거 성공 하셨나요??
저도 호출 해 보고 싶은데 ㅎㅎㅎ

File: src/main/resources/application.yml (hunk 1 above)
Comment (mingmong0018) [reply src/main/resources/application.yml]: 넵 redis 서버를 컨테이너로 띄웠습니다!

File: src/main/java/com/market/global/config/RedisTokenConfig.java (hunk 3 above)
Comment (mingmong0018) [reply src/main/java/com/market/global/config/RedisTokenConfig.java:21]: redis에서 password를 따로 설정하지 않아서 config로 묶지 않고 간단하게 host와 port만 값을 넘기도록 했습니다
password를 설정하지 않은 이유는 redis 컨테이너를 외부에 노출시키지 않고 컨테이너끼리만 통신하도록 했기때문에
보안상 크게 문제가되지 않을 것 같아서 설정하지 않았습니다

File: src/main/java/com/market/global/properties/RedisProperties.java (hunk 4 above)
Comment (mingmong0018) [reply src/main/java/com/market/global/properties/RedisProperties.java:16]: (위 답글과 동일합니다)
redis에서 password를 따로 설정하지 않아서 config로 묶지 않고 간단하게 host와 port만 값을 넘기도록 했습니다
password를 설정하지 않은 이유는 redis 컨테이너를 외부에 노출시키지 않고 컨테이너끼리만 통신하도록 했기때문에
보안상 크게 문제가되지 않을 것 같아서 설정하지 않았습니다

**Comment by mingmong0018** [comment]
네네 서버에 올렸습니다

url: http://175.45.202.144:8080/users
method: POST
Request Body: {
    "email" : "test@test.com",
    "password" : "testtest1!",
    "name" : "minin",
    "tel" : "01039038322"
}
로 테스트 하시면 될 것 같습니다 !

File: src/main/java/com/market/global/config/RedisTokenConfig.java (hunk 3 above)
Comment (f-lab-bradley) [reply src/main/java/com/market/global/config/RedisTokenConfig.java:21]: 어떤 방식으로 설정을 했나요???

**Review by f-lab-bradley — APPROVED** [review]

**Review by f-lab-bradley — APPROVED** [review]
좋은 방법이 있으면 끊임 없이 배포 방법을 변경 하세요 .
Jenkins 에서 git action 으로 넘어왔을 때 어땠어요???
장단점을 잘 정리해서 어디에 저장 해 두세요 ㅎㅎ

**Pull request merged**